.github/workflows/tests.yml

```diff
@@ -0,0 +1,32 @@
+name: Test Go
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.20.3
+
+    - name: Build
```

Comment:
There is no need to build if you're not using that. 
Also I don't see a reason why to waste GH workers on pull request/push. If you wanna set this action I would make optional just as we run deployments